src/Analysis/Ast/Impl/Extensions/PythonModuleExtensions.cs

```diff
@@ -26,5 +28,23 @@ internal static PythonAst GetAst(this IPythonModule module)
             => ((IAstNodeContainer)module).GetAstNode<T>(o);
         internal static void AddAstNode(this IPythonModule module, object o, Node n)
             => ((IAstNodeContainer)module).AddAstNode(o, n);
+
+        /// <summary>
+        /// Returns the string line corresponding to the given location
+        /// </summary>
+        /// <param name="line">The line number</param>
+        internal static string GetLine(this PythonModule module, int line) {
+            SourceLocation source = new SourceLocation(line, 1);
+            string content = module.Content;
+            var start = GetAst(module).LocationToIndex(source);
+            var length = 0;
+
+            if (!content.IsNullOrEmpty()) {
```

Comment:
Two things:
- Use `string.IsNullOrEmpty(content)` instead; the extension method is on `IEnumerable` and is likely less efficient.
- I'd recommend writing `if (string.IsNullOrEmpty(content)) { return content }` instead, to avoid calling Substring on an empty (or worse, null) string.

Reply:
Good points, completely forgot about the latter.